src/Compilers/Core/Portable/CommandLine/CommonCompiler.CompilerEmitStreamProvider.cs

```diff
@@ -114,7 +115,15 @@ private Stream OpenFileStream()
             private void ReportOpenFileDiagnostic(DiagnosticBag diagnostics, Exception e)
             {
                 var messageProvider = _compiler.MessageProvider;
-                diagnostics.Add(messageProvider.CreateDiagnostic(messageProvider.ERR_CantOpenFileWrite, Location.None, _filePath, e.Message));
+                var lockedBy = FileLockCheck.TryGetLockingProcessInfos(_filePath);
+
+                var message = lockedBy.IsEmpty
+                    ? e.Message
+                    : string.Format(CodeAnalysisResources.ExceptionMessage_FileMayBeLockedBy,
+                        e.Message,
+                        string.Join(", ", lockedBy.Select(info => $"'{info.applicationName}' ({info.processId})")));
```

Comment:
This is likely to create a localization problem. I think creating a `DiagnosticInfo` instead would work.

Reply:
Note that msbuild does this the same way: https://github.com/dotnet/msbuild/blob/6cd445d84e59a36c7fbb6f50b7a5a62767a6da51/src/Utilities/LockCheck.cs#L274

Reply:
>What kind of loc problem?

Given the code, different parts of the message can use different language. ```CodeAnalysisResources.ExceptionMessage_FileMayBeLockedBy``` uses language that is current at the time this code is executed, not the language requested when ```Message``` is requested. Strings should not be pulled manually from resources at the time a DiagnosticInfo is created. They should be pulled when one requests the message string and the process should respect the language requested at that time. Basically, the same DiagnosticInfo instance should be able to give a message string for any supported localization language.  

>Not sure what do you suggest to do with DiagnosticInfo.

We support a ```DiagnosticInfo``` instance as an argument for another ```DiagnosticInfo```. I am suggesting to create a ```DiagnosticInfo``` for the "file may be locked by" part instead of accessing resource strings manually.

Reply:
We actually have another helper type that can be used in this specific situation - ```LocalizableResourceString```

Reply:
This is already the case though. We take the exception message from the runtime and include it in the info as an argument.

Reply:
>  I am suggesting to create a DiagnosticInfo for the "file may be locked by" part instead of accessing resource strings manually.

I'll try to do that.

Reply:
Done.